Spanner/src/Connection/Grpc.php

```diff
@@ -120,11 +122,25 @@ class Grpc implements ConnectionInterface
      */
     private $longRunningGrpcClients;
 
+    private function enableConnectionManagement($conf_path)
+    {
+        // TODO(ddyihai): move this function to GrpcTrait, if we are going
+        // to enable the grpc-gcp library for other apis.
+        $conf = new ApiConfig();
+        $conf->mergeFromJsonString($string = file_get_contents($conf_path));
+        $config = new Config($conf);
+        return $config;
+    }
+
     /**
      * @param array $config [optional]
      */
     public function __construct(array $config = [])
     {
+        if (getenv("ENABLE_GCP_OPTIMIZER") === false) {
```

Comment:
To verify, we are enabling this by default and if users don't have semaphores enabled we then throw an exception? If that is the case, I would love if we could discuss alternatives.

Reply:
You are right. How about not using this library when semaphores is not enabled?

Reply:
Hi @dwsupplee .
Thanks for pointing it out. I create a [PR](https://github.com/GoogleCloudPlatform/grpc-gcp-php/pull/24) to disable the gcp library instead of throwing an exception. What do you think about it?

Reply:
I actually don't mind the `Config` class throwing an exception. That will be helpful information for users - WDYT about us checking for semaphore support here in order to determine whether or not to default to using the optimizer?

Reply:
Done. Thanks!
Its' true that throwing an exception provides more information, otherwise it is hard to debug sometimes. Checking semaphore here sounds a great idea!
I added one `if` here to check whether semaphore is enabled. It seems for the windows test, when php_version>7.0, it is not enabled.